ide/wg_Find.ml

```diff
@@ -137,15 +140,36 @@ class finder name (view : GText.view) =
       match found with
       | None ->
         if not backward && not (starti#equal view#buffer#start_iter) then
+          let _ = Ideutils.flash_info "Wrapping..." in
           self#find_from backward view#buffer#start_iter
         else if backward && not (starti#equal view#buffer#end_iter) then
+          let _ = Ideutils.flash_info "Wrapping backwards..." in
           self#find_from backward view#buffer#end_iter
         else
+          let _ = Ideutils.flash_info "String not found" in
           self#set_not_found ()
       | Some (start, stop) ->
+        let text = view#buffer#start_iter#get_text ~stop:view#buffer#end_iter in
+        let rec find_all offs accum =
+          try
+            let i = Str.search_forward self#regex text offs in
+            let j = Str.match_end () in
+            find_all (j + 1) (i :: accum)
+          with Not_found -> List.rev accum
+        in
+        let occurs = find_all 0 [] in
+        let num_occurs = List.length occurs in
+        (* assoc table of offset, occurrence index pairs *)
+        let occur_tbl = List.mapi (fun ndx occ -> (occ,ndx+1)) occurs in
         let _ = view#buffer#select_range start stop in
         let scroll = `MARK (view#buffer#create_mark stop) in
         let _ = view#scroll_to_mark ~use_align:false scroll in
+        let _ =
+          try
+            let occndx = List.assoc start#offset occur_tbl in
+            Ideutils.flash_info ("Found " ^ string_of_int occndx ^ " of " ^ string_of_int num_occurs)
```

Comment:
I think this should say something like "occurrence 3 out of 10" rather than "found 3 of 10", but my macaroni english may trick me here.

Reply:
To my American ears, "occurrence" is a bit formal-sounding. The operation is "Find", so "Found" fits. Others may disagree.